screens/PokemonList.tsx

```diff
@@ -31,11 +35,20 @@ export default function PokemonList() {
     return id;
   };
 
+  const goToMoreCallback = useCallback(
+    pokemonName => {
+      navigation.navigate('PokemonFromTheList', { pokemonName });
+    },
+    [navigation]
+  );
+
   const renderItem = ({ item }: { item: Pokemon }) => {
     const id = retrieveIdFromUrl(item.url);
-
     return (
-      <Text style={styles.pokemonListElement}>
+      <Text
+        style={styles.pokemonListTextElement}
+        onPress={() => goToMoreCallback(item.name)}
```

Comment:
So time for some #performanceTesting 🔥 🚒 

Try to scroll down your Pokemon list fast and let's load e.g. 700 pokemons.
Then scroll up and down fast -> what happens? (feel free to write about it to me on Slack 😄)

Now, we cannot remove this issue easily, we'd have to use a library with native list component and it's not that easy to do that with Expo (and generally this problem is not that easy to solve in 100%). However, we can improve the list rendering greatly by just using good React practices (similar issue could happen on Web).

Right now you're always creating a new function when you use `renderItem`, because you write `() => goToMoreCallback(item.name)` and that creates a new arrow function and then passes it to component down (with "item" saved in scope).

Now, it's a problem. We don't really want to render this component more than once, because we don't sort Pokemons or anything, so they are always in the same place on the list (e.g. index 10) with the same `item`.

What we should do here is extract this `Text` component to a separate function, let's say `PokemonRow` and pass to it `pokemonName` and `pokemonId` -> these two are strings, so they will always be the same when we pass them down and compare (remember that objects might not be the same => `{ id: 5 } === { id: 5 }` returns `false`, because the content of the object is the same, but the object itself is a different instance and we compare object references here, not the content).

That's not the end though, as even if we will `useCallback` in our new component this is not the final step of our performance improvements. We should tell React to not even try to rerender the component if the props are the same -> here comes `React.memo` and `PureComponent`.

You can read about both here: https://www.peterbe.com/plog/memo-instead-of-purecomponent
I'd suggesting comparing how both of them work (which makes the list smoother).

Spoiler -> React.memo should be fine here, but it sometimes causes problems https://dmitripavlutin.com/use-react-memo-wisely/ and PureComponent is a more "no-brainer" option.

You should be also able to test your performance in Expo by using chrome debugger -> https://reactnative.dev/docs/performance#profiling If you'd want to do that then let me know on Slack as it might be a bit hard first time